docs/root/configuration/http/http_filters/router_filter.rst

```diff
@@ -374,6 +374,16 @@ compared to network latency between client and Envoy. This header is set on resp
 
 .. _config_http_filters_router_x-envoy-overloaded_set:
 
+x-envoy-upstream-stream-timeout-ms
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This value is used to configure max stream living duration. It is also similar to
+:ref:`max_stream_duration <envoy_v3_api_field_config.core.v3.HttpProtocolOptions.max_stream_duration>` but
+that is configured with per-stream. If request header has this field, that will be prior to
```

Comment:
> What does "configured per stream" mean in this context? I would have thought that the new header being added here would be per-stream in the sense that it only applied to the stream which the header was sent on. Is that correct?

Yes. Added header works only single stream. It only setup stream duration timer on `UpstreamRequest`. So won't affect subsequent streams.

Reply:
Oh, I see. Ok, so the header configures the timeout for the particular stream which has that header. However, the config option applies to all streams on the route. Is that right? If so, how about:
```
x-envoy-upstream-stream-duration-ms
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This value is used to configure the maximum stream lifetime for the stream which
has this header. If the stream exceeds this lifetime, it will be reset and a 408 response
will be sent to downstream. If the value of the header is 0, then the lifetime will be
infinite and no limit will be enforce. It is similar to
:ref:`max_stream_duration <envoy_v3_api_field_config.core.v3.HttpProtocolOptions.max_stream_duration>`, 
but that configuration applies to all streams on the route. If set, this header will
override the cluster configuration.  

.. _config_http_filters_router_x-envoy-upstream-stream-duration-ms:
```